Radicle-cli: 0.6.1 -> 1.0.0-rc8

roblabla opened the pull request:
## Description of changes

Updates radicle-cli to the latest version.

I unfortunately had to disable tests, as nearly half of them failed in hard to debug ways.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

roblabla committed:
radicle-cli: 0.6.1 -> 1.0.0-rc8

roblabla committed:
radicle-cli: License changed from gpl3+ to asl20 OR mit

roblabla committed:
radicle-cli: Disable tests, some of them refuse to pass

amesgen commented:
FTR: https://github.com/NixOS/nixpkgs/pull/244417#issuecomment-1900884859

roblabla commented:
> FTR: [#244417 (comment)](https://github.com/NixOS/nixpkgs/pull/244417#issuecomment-1900884859)

I'm not sure why this old MR got closed. Version 0.9 is compatible with the current network from my minimal testing. Although radicle's installation script installs 1.0.0-rc2 that version requires Rust 1.77, which hasn't been packaged in nixpkgs yet, so I held off from it.

If we want the latest rc version, we can hold off until https://github.com/NixOS/nixpkgs/pull/298340 lands, and I can update this MR to package 1.0.0-rc2.

koalalorenzo commented:
Do we have any updates on this? Will it make the cut for 24.05?

amesgen commented:
I think updating this to the latest rc version as mentioned by @roblabla seems fine.

roblabla commented:
I updated the MR to use 1.0.0-rc8. Upstream also deprecated their github-based mirror (https://github.com/radicle-dev/heartwood is archived), so I changed it to fetchgit from radicle's official repository (rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5) on app.radicle.xyz

lorenzleutgeb commented on pkgs/applications/version-management/radicle-cli/default.nix:
```diff
@@ -18,28 +18,15 @@
 rustPlatform.buildRustPackage rec {
   pname = "radicle-cli";
-  version = "0.6.1";
+  version = "1.0.0-rc8";
```
```suggestion
  version = "1.0.0-rc.8";
```

For consistency with upstream.

lorenzleutgeb reviewed (changes requested):
Just was pointed at this by @amesgen, and indeed I think it's a pity that I duplicated your work. Sorry about that. However, please check out https://github.com/NixOS/nixpkgs/pull/309050#issuecomment-2094823274 and let's agree on how to proceed. I am happy to first PR against `roblabla:radicle-cli-0.9.0`.

lorenzleutgeb commented:
Binaries produced by this derivation do not report the version like upstream expects (e.g. when reporting a bug).

```
$ ./result/bin/rad version
rad pre-release (unknown)
$ ./result/bin/rad debug | jq -r '.radVersion'
pre-release
```

My conflicting PR has a fix for this, [setting `RADICLE_VERSION`](https://github.com/NixOS/nixpkgs/pull/309050/files#diff-0629dcd6451247ef192e1aa0eb390a6d8ce64612b8974870e1b900ef91d1c47dR14).

lorenzleutgeb commented on pkgs/applications/version-management/radicle-cli/default.nix:
```diff
@@ -18,28 +18,15 @@
-  auditable = false;
+  cargoSha256 = "sha256-Ej9cxuHxiQDwCXOwcIyfCau9Thhz+hf3IevN6F4rDjM=";
 
   nativeBuildInputs = [
```
Did you verify that all `nativeBuildInputs` are still required? I believe that this will build fine without CMake, see #309050.

roblabla commented:
Your MR looks better, let's close this one and converge on #309050
